driver-core/src/test/unit/com/mongodb/internal/connection/BulkWriteBatchCombinerSpecification.groovy

```diff
@@ -47,13 +48,15 @@ class BulkWriteBatchCombinerSpecification extends Specification {
     def 'should get correct result for an insert'() {
         given:
         def combiner = new BulkWriteBatchCombiner(new ServerAddress(), true, ACKNOWLEDGED)
-        combiner.addResult(BulkWriteResult.acknowledged(INSERT, 1, 0, [], []))
+        combiner.addResult(BulkWriteResult.acknowledged(INSERT, 1, 0, [], [new BulkWriteInsert(6, new BsonString('id1'))]))
+        combiner.addResult(BulkWriteResult.acknowledged(INSERT, 1, 0, [], [new BulkWriteInsert(3, new BsonString('id2'))]))
 
         when:
         def result = combiner.getResult()
 
         then:
-        result == BulkWriteResult.acknowledged(INSERT, 1, 0, [], [])
+        result == BulkWriteResult.acknowledged(INSERT, 2, 0, [],
+                [new BulkWriteInsert(3, new BsonString('id2')), new BulkWriteInsert(6, new BsonString('id1'))])
```

Comment:
This test just mirrors the upsert testing and ensures the ordering is as expected.